tensorpack/decomposition.py

```diff
@@ -56,7 +58,7 @@ def perform_PCA(self, flattenon=0):
         self.PCAR2X = [calcR2X(c, mIn=flatData) for c in recon]
         self.sizePCA = [sum(flatData.shape) * rr for rr in self.rrs]
 
-    def Q2X_chord(self, drop=5, repeat=3, mode=0):
+    def Q2X_chord(self, drop=5, repeat=3, mode=0, callback=None, rr_callback=1):
```

Comment:
Why does this function need the callbacks?

If every function needs these callback arguments, it would be better to pass them in within `__init__()`.

Reply:
I'm not certain if every method will be able to handle a callback, so I wanted to be able to control directly from the Decomposition object.

Reply:
If a method can't handle a callback, then we won't have a way to peer inside, and any callback would have no purpose.